test/gbdt/test_gbdt.py

```diff
@@ -7,6 +11,8 @@
 from torch_frame.gbdt import CatBoost, LightGBM, XGBoost
 from torch_frame.testing.text_embedder import HashTextEmbedder
 
+TEST_DIR = tempfile.TemporaryDirectory()
```

Comment:
Can we use `with` statement for better context management. If exception occurs inside the code, then the tempdir might not be deleted.

Reply:
Ah actually, it won't clean up.

Reply:
Also fixed other uses of `tmp_path` as part of this PR!